### ISSUE TITLE
XML in campfire lines are not valiad

### ISSUE THREAD

**Issue by minecraftchest1** [basecamp/bc3-api#326]
The image tag in the XML for mentions in Basecamp 3 mentions is not closed. This makes it hard to parse the line text and get the content I am interested in, as the parser I use refuses to parse invalid XML for what I feel are obvious reasons. I should not be forced to try parsing it from scratch because your your software doesn't work correctly.

![image](https://user-images.githubusercontent.com/42992675/156379568-b013d2c4-fb72-4de3-b136-091b874e90e6.png)
![image](https://user-images.githubusercontent.com/42992675/156379955-5ab29ec7-9bc9-441c-af71-26cc9d891e06.png)

``` XML
<bc-attachment
  sgid="BAh7CEkiCGdpZAY6BkVUSSIpZ2lkOi8vYmMzL1BlcnNvbi8zNzM1OTM0OD9leHBpcmVzX2luBjsAVEkiDHB1cnBvc2UGOwBUSSIPYXR0YWNoYWJsZQY7AFRJIg9leHBpcmVzX2F0BjsAVDA=--a7680e74051326fe39233ab33026741d5c4ece25"
  content-type="application/vnd.basecamp.mention">
  <figure>
    <img
      srcset="https://bc3-production-assets-cdn.basecamp-static.com/5230042/people/BAhpBPQOOgI=--a76b8f304bfd22e6aa98643582efed2e44bd53e1/avatar-20-x1?v=1 1x,
        https://bc3-production-assets-cdn.basecamp-static.com/5230042/people/BAhpBPQOOgI=--a76b8f304bfd22e6aa98643582efed2e44bd53e1/avatar-20-x2?v=1
        2x, https://bc3-production-assets-cdn.basecamp-static.com/5230042/people/BAhpBPQOOgI=--a76b8f304bfd22e6aa98643582efed2e44bd53e1/avatar-20-x3?v=1 3x,
        https://bc3-production-assets-cdn.basecamp-static.com/5230042/people/BAhpBPQOOgI=--a76b8f304bfd22e6aa98643582efed2e44bd53e1/avatar-20-x4?v=1 4x"
      title="Taylor Whisenant, Teacher Sponsor at Athens Renaissance Robotics" alt="Taylor Whisenant"
      data-avatar-for-person-id="37359348"
      data-controller="lazy-image" class="avatar"
      src="https://bc3-production-assets-cdn.basecamp-static.com/5230042/people/BAhpBPQOOgI=--a76b8f304bfd22e6aa98643582efed2e44bd53e1/avatar-20-x1?v=1"
      width="20" height="20">
    <figcaption>
      Taylor
    </figcaption>
  </figure>
</bc-attachment>
am-3947 looks to be a 3/8" size - would recommend confirming with
<bc-attachment
  sgid="BAh7CEkiCGdpZAY6BkVUSSIpZ2lkOi8vYmMzL1BlcnNvbi8zNzM2MTk0ND9leHBpcmVzX2luBjsAVEkiDHB1cnBvc2UGOwBUSSIPYXR0YWNoYWJsZQY7AFRJIg9leHBpcmVzX2F0BjsAVDA=--41350a7a18681cab321dd675e52995de5621bd5d"
  content-type="application/vnd.basecamp.mention">
  <figure>
    <img
      srcset="https://bc3-production-assets-cdn.basecamp-static.com/5230042/people/BAhpBBgZOgI=--df6599bcf0d0145132ddd504c63653b6d25614c9/avatar-20-x1?v=1 1x,
      https://bc3-production-assets-cdn.basecamp-static.com/5230042/people/BAhpBBgZOgI=--df6599bcf0d0145132ddd504c63653b6d25614c9/avatar-20-x2?v=1 2x,
      https://bc3-production-assets-cdn.basecamp-static.com/5230042/people/BAhpBBgZOgI=--df6599bcf0d0145132ddd504c63653b6d25614c9/avatar-20-x3?v=1 3x,
      https://bc3-production-assets-cdn.basecamp-static.com/5230042/people/BAhpBBgZOgI=--df6599bcf0d0145132ddd504c63653b6d25614c9/avatar-20-x4?v=1 4x"
      title="Ian Collier" alt="Ian Collier" data-avatar-for-person-id="37361944" data-controller="lazy-image" class="avatar"
      src="https://bc3-production-assets-cdn.basecamp-static.com/5230042/people/BAhpBBgZOgI=--df6599bcf0d0145132ddd504c63653b6d25614c9/avatar-20-x1?v=1" width="20" height="20">
      <figcaption>
        Ian
      </figcaption>
  </figure>
</bc-attachment>
that was his intent. Might be easier to get one of the "assorted kits" am-4071, -4072, or -4073 rather than buying piecemeal  
```
```
~$ jq '.[0].content' basecamp-campfire-1.json --raw-output | xmllint -
-:6: parser error : Opening and ending tag mismatch: img line 2 and figure
</figure></bc-attachment> am-3947 looks to be a 3/8" size - would recommend conf
         ^
-:6: parser error : Opening and ending tag mismatch: figure line 1 and bc-attachment
</figure></bc-attachment> am-3947 looks to be a 3/8" size - would recommend conf
                         ^
-:11: parser error : Opening and ending tag mismatch: img line 7 and figure
</figure></bc-attachment> that was his intent. Might be easier to get one of the
         ^
-:11: parser error : Opening and ending tag mismatch: figure line 6 and bc-attachment
</figure></bc-attachment> that was his intent. Might be easier to get one of the
                         ^
-:12: parser error : Premature end of data in tag bc-attachment line 6

^
~$ 
```

**Comment by ramhoj** [comment]
I'm sorry for the very delayed response to this issue @minecraftchest1.

Based on your terminal output above it looks like you're parsing the response with `xmllint`. The response above is HTML (and not XML). `xmllint` has an `--html` option but it doesn't support [custom elements](https://html.spec.whatwg.org/multipage/custom-elements.html#custom-elements).

You can verify that the markup is valid by pasting it into https://validator.w3.org/nu/#textarea. If you want to parse it from the command line you can do so with [tidy](https://www.html-tidy.org/). E.g `tidy --new-blocklevel-tags bc-attachment a_file.html`.
